server/page.js

```diff
@@ -1,10 +1,41 @@
-function page (contents, dehydratedAppState) {
+const serializeJS = require('serialize-javascript')
+
+function page (contents, appState) {
+  var dehydratedAppState = serializeJS(appState)
+
+  function renderMetaTags () {
+    var s = ''
+    var md = appState.archive && appState.archive.metadata ? appState.archive.metadata : null
+    if (!md) return s
+    if (md.title) {
```

Comment:
then we split the logic either at the `renderMetaTags()` or `page.js` level. the point is that the page should always be rendered from the same `dehydratedAppState` object that is used to render page.js with as well as rehydrating the app, otherwise we'll end up with weird inconsistencies down the line.